mint-auditor/src/convert/gnosis_safe_deposit.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Convert to/from mc_mint_auditor_api::GnosisSafeDeposit.
+
+use crate::{
+    db::GnosisSafeDeposit as DbGnosisSafeDeposit,
+    gnosis::{EthAddr, EthTxHash},
+    Error,
+};
+use mc_mint_auditor_api::GnosisSafeDeposit as ProtoGnosisSafeDeposit;
+use std::str::FromStr;
+
+/// Convert DbGnosisSafeDeposit --> ProtoGnosisSafeDeposit
+impl From<&DbGnosisSafeDeposit> for ProtoGnosisSafeDeposit {
+    fn from(src: &DbGnosisSafeDeposit) -> Self {
+        let mut dst = Self::new();
+        dst.set_id(src.id().unwrap_or_default());
+        dst.set_eth_tx_hash(src.eth_tx_hash().to_string());
+        dst.set_eth_block_number(src.eth_block_number());
+        dst.set_safe_addr(src.safe_addr().to_string());
+        dst.set_token_addr(src.token_addr().to_string());
+        dst.set_amount(src.amount());
+        dst.set_expected_mc_mint_tx_nonce_hex(src.expected_mc_mint_tx_nonce_hex().to_string());
+        dst
+    }
+}
+
+/// Convert ProtoGnosisSafeDeposit --> DbGnosisSafeDeposit
+impl TryFrom<&ProtoGnosisSafeDeposit> for DbGnosisSafeDeposit {
+    type Error = Error;
+
+    fn try_from(src: &ProtoGnosisSafeDeposit) -> Result<Self, Self::Error> {
+        Ok(Self::new(
+            if src.get_id() == 0 {
+                None
+            } else {
+                Some(src.get_id())
+            },
```

Comment:
I was curious about this, so checked in godbolt and it appears that with optimization enabled the compiler is able to optimize this away, but as you pointed out get_id does get called twice when optimization is disabled. Your version is also 1 line shorter so that's a win/win